src-ghc/Pact/Types/Crypto.hs

```diff
@@ -196,11 +148,12 @@ instance Scheme (SPPKScheme 'ED25519) where
   type PrivateKey (SPPKScheme 'ED25519) = Ed25519.PrivateKey
   type Signature (SPPKScheme 'ED25519) = Ed25519.Signature
 
-  _sign _ (Hash msg) pub priv = return $ Ed25519.sign msg priv pub
-  _valid _ (Hash msg) pub sig = Ed25519.valid msg pub sig
-  _genKeyPair _ = ed25519GenKeyPair
-  _getPublic _ = Just . ed25519GetPublicKey
-  _formatPublicKey _ p = toBS p
+  -- TODO: Check this with the right compiler flags.
```

Comment:
what compiler flag checks do we need?

Reply:
This one from `pact.cabal`:

```
flag cryptonite-ed25519
  description: use crypton instead of ed25519-donna
  default:     True
  manual:      True
```

But, when I toggle it off, I see lots of compiler errors for code not related to my changes. I think we should fix up that particular CPP path later.